src/canvas-extractor/edfi_canvas_extractor/graphql/extractor.py

```diff
@@ -0,0 +1,181 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import json
+import logging
+import os
+import requests
+
+from dotenv import load_dotenv
+from typing import Dict, List
+
+from .schema import query_builder
+from .utils import validate_date
+
+
+class Singleton(object):
+    _instance = None
+
+    def __new__(class_, *args, **kwargs):
+        if not isinstance(class_._instance, class_):
+            class_._instance = object.__new__(class_, *args, **kwargs)
+        return class_._instance
+
+
+class GraphQLExtractor(Singleton):
+    courses: List
+    enrollments: List
+    sections: List
+    students: List
+
+    def __init__(self):
+        self.data = None
+        self.courses = list()
+        self.sections = list()
+        self.students = list()
+        self.enrollments = list()
+
+    def get_from_canvas(self, query: str) -> Dict:
+        """
+        Get GraphQL Query from Canvas
+
+        Parameters
+        ----------
+        query: str
+            query string for GraphQL
+
+        Returns
+        -------
+        Dict JSON Object
+        """
+
+        load_dotenv()
+        CANVAS_URL = os.getenv('CANVAS_BASE_URL')
+        CANVAS_TOKEN = os.getenv('CANVAS_ACCESS_TOKEN')
```

Comment:
These values are parsed from the command line at runtime and should be passed as arguments, rather than as environment variables. See the `MainArguments` object passed to the `run()` function.